src/Jeffijoe.MessageFormat.Tests/Formatting/Formatters/NumberFormatterTests.cs

```diff
@@ -0,0 +1,65 @@
+using Xunit;
+
+namespace Jeffijoe.MessageFormat.Tests.Formatting.Formatters
+{
+    public class NumberFormatterTests
+    {
+        [Theory]
+        [InlineData(69, "69.000")]
+        [InlineData(69.420, "69.420")]
+        [InlineData(123_456.789, "123,456.789")]
+        public void NumberFormatter_Decimal(decimal number, string expected)
+        {
+            var mf = new MessageFormatter(locale: "en-US");
+            var actual = mf.FormatMessage("{value, number }", new
+            {
+                value = number
+            });
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData(0.2, "20.000%")]
+        [InlineData(1.2, "120.000%")]
+        public void NumberFormatter_Percent(decimal number, string expected)
+        {
+            var mf = new MessageFormatter(locale: "en-US");
+            var actual = mf.FormatMessage("{value, number,percent}", new
+            {
+                value = number
+            });
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData("en-US", 20, "$20.00")]
+        [InlineData("en-US", 99.99, "$99.99")]
+        [InlineData("da-DK", 99.99, "99,99 kr.")]
+        public void NumberFormatter_Currency(string locale, decimal number, string expected)
+        {
+            var mf = new MessageFormatter(locale: locale);
+            var actual = mf.FormatMessage("{value, number, currency }", new
+            {
+                value = number
+            });
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData(20,"#0.0", "20.0")]
+        [InlineData(99.234, "#0.0", "99.2")]
```

Comment:
@glen-84 you mentioned that skeletons have their own pattern symbols, which means we wouldn't be able to use .NET format specifiers like these, correct?

If that is the case, then I am somewhat conflicted. On one hand, I don't want to implement the entire ICU spec (@NightOwl888 seems to be doing this in ICU4N), but on the other, I do like the flexibility of being able to be a bit more specific with the format to use.

Reply:
I [commented on](https://github.com/jeffijoe/messageformat.net/issues/17#issuecomment-1409091444) date skeletons. Number skeletons are also [quite different](https://unicode-org.github.io/icu/userguide/format_parse/numbers/skeletons.html#examples).

Reply:
Both of these require locale data, right?

What do you think the best course of action is for the short term here? Support using .NET format specifiers in the skeleton for now, or drop skeleton support entirely?

Reply:
The [documentation](https://unicode-org.github.io/icu/userguide/format_parse/numbers/skeletons.html#overview) states that number skeletons are locale-agnostic, but it seems like it would be a lot of work to support all/most of the features.

Since this library is ICU-related, I think it would be confusing for it to support .NET format specifiers instead of ICU number skeletons.

For these reasons, as I mentioned before, it might be best not to support skeleton syntax at this time.

Reply:
@glen-84  Should we throw when detecting skeletons instead? Would that be better or worse than attempting to format using .NET format specifiers? I am not sure how many (or if any) of them are similar to ICU ones, but I am inclined to at least not throw, however, I also understand that it may be harder to potentially spot these issues for an end-user of the library.

Reply:
I would suggest throwing, and documenting that skeleton syntax is not supported.

Reply:
I thought about it some more, and given that we will only be supporting a very small subset for now, I think just using the default format is safer than throwing.

Reply:
So if the user does:

```
{0, number, :: +! K currency/GBP}
```

You'll silently ignore the skeleton? I'm not sure if that's ideal. It would also be a breaking change if someone _did_ decide to add support for this in the future.

Reply:
@glen-84 Yes, that is also a valid point. And I suppose we already throw now without the support this PR is adding. Ok, you've convinced me. I'll throw when detecting an unsupported pattern unless it's a format string like `{value, number, 0.##}` or `{value, date, dd-MM-yyy}`.

Reply:
I'm not sure if you should support format strings either. 😅

The problem is that:

- They're not the same as ICU format strings.
- ICU message format accepts ICU format strings, not .NET format strings.

I think that this will just confuse the user.

I still think that [this](https://github.com/jeffijoe/messageformat.net/issues/17#issuecomment-1409091444) is the best option for now ([potential API](https://github.com/jeffijoe/messageformat.net/issues/17#issuecomment-1417824861)), supporting [predefined styles](https://unicode-org.github.io/icu/userguide/format_parse/messages/#predefined-styles-recommended) along with a way to override the format strings per style and locale.

Reply:
I could be wrong, but it looks like most of the basic tokens are the same, with the exception of AM/PM (ICU `a`, .NET `tt`)? 

I think supporting those for the time being would be good enough, as it allows enough flexibility to not require some external customization API. I will still support the predefined styles as we discussed for convenience, but I think adding an extra layer of customization might be another can of worms; where do we draw the line?

Reply:
[String patterns are discouraged](https://unicode-org.github.io/icu/userguide/format_parse/messages/#string-patterns-discouraged), because they're not localized. I think that by supporting them, you'd be encouraging poor localization practices.

Supporting predefined styles like `medium` allows them to be mapped to a locale-specific pattern, which can default to a .NET format specifier like `d` (which itself is culture-specific).

Since it's not always a perfect match, it may well be necessary to customize this default .NET format string. This could _maybe_ come later, but it might be quite limiting without it.

Reply:
Wouldn't overriding `medium` etc also be discouraged since you could be using non-standard localizations?

Reply:
That way, if you decide you want to allow the use of .NET format specifiers, then you have an avenue for enabling it without the library explicitly encouraging it.

Reply:
I think it's okay. It just means that there's only one method body for all cultures, so it might be more difficult to split that logic. The formatter would need to somehow pull the data for all supported cultures from some source, but perhaps it could be loaded from a file (for example).